sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -175,7 +187,7 @@ public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse
         this.fluxHandle = asyncPollRequestWithDelay()
             .flux()
             .repeat()
-            .takeUntil(pollResponse -> hasCompleted())
+            .takeUntil(pollResponse -> isComplete())
             .share()
             .delaySubscription(onActivation);
```

Comment:
One thing I was wondering is if the onActivation fails, should we keep polling? delaySubscription only delays it.